CONFIGURATION.md

```diff
@@ -50,15 +52,17 @@ The following configuration options are complementary to the RdKafka [options](h
 | internal.producer.timeout.ms | \>= -1 | 1000 | Sets the timeout on poll and flush operations. Set to **-1** for infinite timeout. |
 | internal.producer.retries | \>= 0 | 0 | Sets the number of times to retry sending a message from the internal queue before giving up. Note that this setting is independent of the rdkafka setting **message.send.max.retries**. This setting applies to each individual message and not to entire batches of messages like **message.send.max.retries** does. |
 | internal.producer.payload.policy | passthrough, copy | copy | Sets the payload policy on the producer. For sync producers, use **passthrough** for maximum performance. |
-| internal.producer.preserve.message.order | true, false | false | Set to **true** if strict message order must be preserved. When setting this, rdkafka option **max.in.flight** will be set to **1** to avoid possible reordering of packets. Messages will be buffered and sent sequentially, waiting for broker acks before the next one is processed. This ensures delivery guarantee at the cost of performance. The internal buffer queue size can be checked via `ProducerMetadata::getInternalQueueLength()`. An intermediate, solution with better performance would be to set this setting to **false** and set **max.in.flight=1** which will skip internal buffering altogether. This will guarantee ordering but not delivery. |
-| internal.producer.max.queue.length | \> 0 | 10000 | Maximum size of the internal queue when producing asynchronously with strict order preservation. |
-| internal.producer.wait.for.acks.timeout.ms | \>= -1 | 0 | If set, then the producer will wait up to the timeout specified when producing a message via `send()` or `post()`. Set to **-1** for infinite timeout. |
-| internal.producer.flush.wait.for.acks.timeout.ms | \>= -1 | 0 | If set, then the producer will wait up to the timeout specified when flushing the internal queue. Set to **-1** for infinite timeout. |
+| internal.producer.preserve.message.order | true, false | false | Set to **true** if strict message order must be preserved. When setting this, rdkafka option library will also **max.in.flight=1** and **message.send.max.retries=1** to avoid possible reordering of packets, unless **enable.idempotence=true** or if the application explicitly overwrites them. Messages will be buffered and sent sequentially, waiting for broker acks before the next one is processed. This ensures delivery guarantee at the cost of performance. The internal buffer queue size can be checked via `ProducerMetadata::getInternalQueueLength()`. An intermediate, solution with better performance would be to set this setting to **false** and set **max.in.flight=1** which will skip internal buffering altogether. This will guarantee ordering but not delivery. This setting is somewhat redundant to **enable.idempotence** but offers more flexibility, so both should not be sent concurrently. |
```

Comment:
The second sentence is incoherent. Try something more like:
When this is set to true, to avoid possible reordering of packets, rdkafka options will be set as follows: **max.in.flight=1**, **message.send.max.retries=1**. **enable.idempotence=true**, or the application configuration can override this.

Reply:
Fixed